src/components/Auth/Profile/tabs/Dashboard.jsx

```diff
@@ -1,51 +1,88 @@
+import { useEffect, useState } from "react";
+import { UserApi } from "../../../../api/auth/user";
+
 export default function Dashboard() {
+  const [user, setUser] = useState(null);
+
+  useEffect(() => {
+    const fetchUser = async (id) => {
+      try {
+        const { data: userDetails } = await UserApi(id);
+        setUser(userDetails);
+      } catch (error) {
+        console.error("Error fetching user data:", error);
+      }
+    };
+
+    const token = localStorage.getItem("token");
+
+    if (!token) {
+      console.log("No token found");
+      return;
+    }
+
+    try {
+      const payload = token.split(".")[1];
+      const decodedPayload = atob(payload);
+      const decodedPayloadObject = JSON.parse(decodedPayload);
+
+      fetchUser(decodedPayloadObject?.id);
+    } catch (error) {
+      console.error("Error decoding token:", error);
+    }
+  }, []);
+
+  const formatHealthStatus = (healthStatus) => {
+    if (!healthStatus) return "Loading...";
+    
+    const deficiencies = Object.entries(healthStatus)
+      .filter(([_, value]) => value === true)
+      .map(([key]) => key);
+
+    return deficiencies.length > 0 ? deficiencies.join(", ") : "No deficiencies reported";
+  };
+
   return (
     <>
       <div className="welcome-msg w-full">
         <div>
-          <p className="text-qblack text-lg">Hello, John Doe</p>
+          <p className="text-qblack text-lg">
+            Hello, {user ? `User ${user.firstName || ""} ${user.lastName || ""}` : "Guest"}
+          </p>
           <h1 className="font-bold text-[24px] text-qblack">
             Welcome to your Profile
           </h1>
         </div>
       </div>
+
       <div className="dashboard-info mt-8 flex flex-col bg-primarygray px-7 py-7">
         <div>
-          <p className="title text-[22px] font-semibold">
-            Personal Information
-          </p>
+          <p className="title text-[22px] font-semibold">Personal Information</p>
           <div className="mt-5">
             <table>
               <tbody>
-                {/* Each row now stacks its cells vertically */}
                 <tr className="mb-5 block">
                   <td className="text-base text-qgraytwo block mb-1">Name:</td>
                   <td className="text-base text-qblack font-medium block">
-                    John Doe
+                    {user ? `${user.firstName} ${user.lastName}` : "Loading..."}
```

Comment:
make a fallback mechanism , in case of something goes wrong, or firstName or lastName does not exsits 
```suggestion
                    {user ? `${user?.firstName || ""} ${user?.lastName || ""}` : "Loading..."}
```

Reply:
btw one note here , if you fail to fetch the user it would keep showing for the user the `Loading...` , so what we do in a real life scenario , we add a loading state to show the loading stuff when we are actually fetching stuff , and if we fail we do a sage mechanism like "Invalid" or empty strings ""

I am saying to change this , but I saying this for you to know and keep into consideration